common/src/main/java/com/revenuecat/purchases/common/ReceiptInfo.kt

```diff
@@ -53,3 +69,38 @@ class ReceiptInfo(
             "duration=$duration)"
     }
 }
+
+internal fun StoreProduct.platformProductId(): PlatformProductId {
+    return PlatformProductId(id)
+}
+
+internal fun SubscriptionOption.platformProductId(): PlatformProductId? {
+    return when (this) {
+        is GoogleSubscriptionOption -> GooglePlatformProductId(
+            productId,
+            basePlanId,
+            offerId
+        )
+        else -> null
+    }
+}
+
+internal open class PlatformProductId(open val productId: String) {
+    open val toMap: Map<String, String?>
+        get() = mapOf(
+            "product_id" to productId
+        )
+}
+
+internal class GooglePlatformProductId(
+    override val productId: String,
+    val basePlanId: String? = null,
+    val offerId: String? = null
+) : PlatformProductId(productId) {
+    override val toMap: Map<String, String?>
```

Comment:
serious nit: since this is a val and not a function, idk if the "to" naming is ideal.. maybe just .asMap